homeassistant/components/lcn/websocket.py

```diff
@@ -0,0 +1,421 @@
+
+from typing import Final
+
+import lcn_frontend as lcn_panel
+import voluptuous as vol
+
+from homeassistant.components import panel_custom, websocket_api
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_ADDRESS,
+    CONF_DEVICES,
+    CONF_DOMAIN,
+    CONF_ENTITIES,
+    CONF_ENTITY_ID,
+    CONF_NAME,
+    CONF_RESOURCE,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    ADD_ENTITIES_CALLBACKS,
+    CONF_DOMAIN_DATA,
+    CONF_HARDWARE_SERIAL,
+    CONF_HARDWARE_TYPE,
+    CONF_SOFTWARE_SERIAL,
+    CONNECTION,
+    DOMAIN,
+)
+from .helpers import (
+    DeviceConnectionType,
+    async_update_device_config,
+    generate_unique_id,
+    get_device_config,
+    get_device_connection,
+    get_resource,
+    purge_device_registry,
+    purge_entity_registry,
+    register_lcn_address_devices,
+)
+from .schemas import (
+    ADDRESS_SCHEMA,
+    DOMAIN_DATA_BINARY_SENSOR,
+    DOMAIN_DATA_CLIMATE,
+    DOMAIN_DATA_COVER,
+    DOMAIN_DATA_LIGHT,
+    DOMAIN_DATA_SCENE,
+    DOMAIN_DATA_SENSOR,
+    DOMAIN_DATA_SWITCH,
+)
+
+URL_BASE: Final = "/lcn_static"
+
+
+async def register_panel(hass: HomeAssistant) -> None:
+    """Register the LCN Panel and Websocket API."""
+    websocket_api.async_register_command(hass, websocket_get_device_configs)
+    websocket_api.async_register_command(hass, websocket_get_entity_configs)
+    websocket_api.async_register_command(hass, websocket_scan_devices)
+    websocket_api.async_register_command(hass, websocket_add_device)
+    websocket_api.async_register_command(hass, websocket_delete_device)
+    websocket_api.async_register_command(hass, websocket_add_entity)
+    websocket_api.async_register_command(hass, websocket_delete_entity)
+
+    if DOMAIN not in hass.data.get("frontend_panels", {}):
+        hass.http.register_static_path(
+            URL_BASE,
+            path=lcn_panel.locate_dir(),
+            cache_headers=lcn_panel.is_prod_build,
+        )
+        await panel_custom.async_register_panel(
+            hass=hass,
+            frontend_url_path=DOMAIN,
+            webcomponent_name=lcn_panel.webcomponent_name,
+            config_panel_domain=DOMAIN,
+            module_url=f"{URL_BASE}/{lcn_panel.entrypoint_js}",
+            embed_iframe=True,
+            require_admin=True,
+        )
+
+
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {vol.Required("type"): "lcn/devices", vol.Required("host_id"): cv.string}
+)
+@websocket_api.async_response
+async def websocket_get_device_configs(
+    hass: HomeAssistant, connection: websocket_api.ActiveConnection, msg: dict
+) -> None:
+    """Get device configs."""
+    if not (config_entry := hass.config_entries.async_get_entry(msg["host_id"])):
+        connection.send_result(msg["id"], False)
+        return
```

Comment:
Let this check happen in a decorator instead. The decorator will then call the decorated function passing the config entry; like this:
```py
@websocket_api.require_admin
@websocket_api.websocket_command(
    {vol.Required("type"): "lcn/devices", vol.Required("host_id"): cv.string}
)
@websocket_api.async_response
@get_config_entry
async def websocket_get_device_configs(
    hass: HomeAssistant,
    connection: websocket_api.ActiveConnection,
    msg: dict,
    config_entry: ConfigEntry,
) -> None:
    """Get device configs."""
    connection.send_result(msg["id"], config_entry.data[CONF_DEVICES])
```

Reply:
Added a decorator.